src/Stack/Types/Build.hs

```diff
@@ -645,9 +645,7 @@ configureOptsNoDir econfig bco deps isLocal package = concat
     -- earlier. Cabal also might do less work then.
     useExactConf = configAllowNewer config
 
-    cabalVersion = envConfigCabalVersion econfig
-    newerCabal = cabalVersion >= $(mkVersion "1.22")
-    newerProfFlagCabal = cabalVersion >= $(mkVersion "1.21.1")
+    newerCabal = envConfigCabalVersion econfig >= $(mkVersion "1.22")
```

Comment:
It certainly looks like an improvement from this POV, but actually, this just reverts the changes I made earlier to check for Cabal >= 1.21.1, because checking for a Cabal development version just isn't really useful. But thanks for the compliment anyway!